### ISSUE TITLE
is there a way to get evaluation metric on a training set?

### ISSUE THREAD

**Issue by yuliaUU** [biomodhub/biomod2#94]
Hi! I wonder if it is possible to extract the evaluation metrics on a training dataste? i use `get_evaluations` function: but that one gives me evaluation metrics like cutoff, sensitivity, specificity and testing.data. But I would like to get the same columns but for training data
![image](https://user-images.githubusercontent.com/54612866/187762570-d9280aa2-a13b-4fc1-a56d-f13e4d61e861.png)

**Comment by yuliaUU** [comment]
I am also a bit confused with this output: if I have a FULL run: I cant have test/train data: since I run all the data. so I am not sure how to interpret Testing.data column

**Comment by MayaGueguen** [comment]
Hello Yulia,

Actually, evaluation metrics are only calculed for _validation lines_. And if no _validation lines_ are available, _calibration lines_ are taken. The column name `Testing.data` might not be the best one, but it was initially set to mean "_values calculated on validation data_".

But I agree that would be best to have both values returned, for calibration AND validation.
I put that into my todo list, and will post a message here as soon as it will be available :information_source: 

Maya

**Comment by yuliaUU** [comment]
Thank you for explanation Maya! Can you please point me out to the documentation: I  am not sure I understand what validation lines or calibration lines  are.

**Comment by MayaGueguen** [comment]
Yes, sorry, I jumped a bit over things... :racehorse: 

Calibration and validation lines correspond to which points are going to be used to calibrate and test your data.
It comes from the `data.split.perc` parameter within the [BIOMOD_Modeling](https://biomodhub.github.io/biomod2/reference/BIOMOD_Modeling.html) function, and you can check the "_nb.rep & data.split.perc_" paragraph into the _Details_ section.

You can access the calibration / validation lines information through the [get_calib_lines](https://biomodhub.github.io/biomod2/reference/getters.out.html) function, applied to your object obtained from [BIOMOD_Modeling](https://biomodhub.github.io/biomod2/reference/BIOMOD_Modeling.html) function.

Hope it's clearer now !
Maya

**Comment by yuliaUU** [comment]
Yes! All clear! thank you

**Comment by yuliaUU** [comment]
just to repot: `get_calib_lines(obj, as.data.frame = FALSE, ...)` when I set `as.data.frame = TRUE`- it does not convert the object to a dataframe
